src/apiServices/ClientService.ts

```diff
@@ -17,29 +41,57 @@ export class ClientService {
         this.toolName = toolName
     }
 
+    /**
+     *  Login to the API using ethAddress and password specified in the library constructor.
+     * @return {Promise<JwtTokensInterface>}  Returns an object containing two tokens (access+refresh) that can be saved in storage.
+     */
     async login(): Promise<JwtTokensInterface> {
         this.jwtTokens = await this.authService.login(this.ethAddress, this.password)
         this.analysesService = new AnalysesService(this.jwtTokens, this.toolName)
-        console.log(this.jwtTokens, 'tokensss')
+
         return this.jwtTokens
     }
 
-    async loginWithToken(jwtTokens: JwtTokensInterface) {
+    /**
+     *  Login to the API using a set of pre-existing tokens
+     *   Can be used when user has previously log in and stored those tokens in memory
+     * @param jwtTokens object containing access + refresh token
+     * - example: loginWithToken({access:'foo', refresh: 'foo2'})
+     * @return {void}
+     */
+    loginWithToken(jwtTokens: JwtTokensInterface) {
         this.analysesService = new AnalysesService(jwtTokens)
     }
 
+    /**
+     *  Logout from the API
+     * @returns Resolves with API response or throw error
+     */
     async logout() {
         return await this.authService.logout()
     }
 
+    /**
+     *   Returns API current version
+     *   Does not require login
+     */
     async getVersion() {
         return await this.authService.getVersion()
     }
 
+    /**
+     *  Refresh current set of tokens.
+     * @param jwtTokens - Object containing access + refresh token
+     * @return {Promise<any>}  Returns new set of tokens or throws error.
+     */
     async refreshToken(jwtTokens?: JwtTokensInterface) {
-        return await this.analysesService.refreshToken(jwtTokens)
+        return await this.authService.refreshToken(jwtTokens)
     }
 
+    /**
+     *   Returns API stats
+     *   Internal only
```

Comment:
Perhaps adding a note here about how it requires admin credentials or similar so that users are not confused would be good.

Reply:
nice one